### ISSUE TITLE
Our Android users are receiving a crash in com.pushio.manager.PIOPermissionsActivity.onRequestPermissionsResult

### ISSUE THREAD

**Issue by shamilovtim** [oracle/pushiomanager-react-native#15]
Hello! 

We need an urgent bugfix to this. Could you please provide an ETA for a resolution? Our users are receiving a crash in:
`java.lang.ArrayIndexOutOfBoundsExceptionPIOPermissionsActivity
length=0; index=0`

Full stack trace:
```
java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
        at com.pushio.manager.PIOPermissionsActivity.onRequestPermissionsResult(PIOPermissionsActivity.java:54)
        at android.app.Activity.requestPermissions(Activity.java:5278)
        at androidx.core.app.ActivityCompat.requestPermissions(ActivityCompat.java:502)
        at com.pushio.manager.PIOPermissionsActivity.processIntent(PIOPermissionsActivity.java:47)
        at com.pushio.manager.PIOPermissionsActivity.onNewIntent(PIOPermissionsActivity.java:22)
        at android.app.Activity.performNewIntent(Activity.java:8213)
        at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1409)
        at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1422)
        at android.app.ActivityThread.deliverNewIntents(ActivityThread.java:4031)
        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:4043)
        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:53)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8512)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

Galaxy S10
Android 11
Version 6.48

**Comment by neerhaj** [comment]
This issue is same as #11 . Fix is available with Android SDK 6.48.1.

**Comment by shamilovtim** [comment]
@neerhaj Correct me if I'm wrong but you had said that the issue in #11 was only present in debug builds. Our production release users are receiving this crash.

**Comment by shamilovtim** [comment]
@neerhaj This ticket is not the same as #11. You asked me in #11 to set location to false, which I did. In this case, this crash is occurring even while location is set to false. Can you please reopen an investigate?

**Comment by neerhaj** [comment]
The stack-traces, both above and in #11, point to the same line/class with the same error. So, the issue is the same.

The core issue is of the app being reloaded when the SDK is in the process of initiating a registration. In #11, we saw this was due to the Hot Reload function in debug mode. Other causes for app reloading while being used (in release mode) could be the Multi-Window/Split-Screen/App resize feature some of the devices have, which causes the system to reload the app when the window size changes.

In any case, the fix done for #11 is applicable for both debug and release builds. 

Based on the crash information, this user had the app with SDK 6.48. This issue was fixed in 6.48.1, so, once the updated app (with SDK 6.48.1) is published and users upgrade to it, this will no longer be an issue.
